shell/src/main/java/alluxio/cli/fs/command/CpCommand.java

```diff
@@ -232,17 +290,26 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
         int numThreads;
         if (cl.hasOption("t")) {
           try {
-            numThreads = (int) cl.getParsedOptionValue("t");
+            numThreads = ((Number) cl.getParsedOptionValue("t")).intValue();
           } catch (ParseException e) {
             throw new IOException("Failed to parse option -t into an integer", e);
           }
         } else {
           numThreads = Runtime.getRuntime().availableProcessors() * 2;
         }
-        CopyThreadPoolExecutor pool = new CopyThreadPoolExecutor(numThreads, System.out,
-            mFileSystem.exists(dstPath) ? null : dstPath);
-        copyFromLocalFileList(pool, srcPaths, dstPath);
-        pool.shutdown();
+        CopyThreadPoolExecutor pool = new CopyThreadPoolExecutor(numThreads, System.out, System.err,
+            mFileSystem, mFileSystem.exists(dstPath) ? null : dstPath);
+        try {
+          createDstDir(dstPath);
+          for (AlluxioURI src : srcPaths) {
+            AlluxioURI dst = new AlluxioURI(dstPath, new AlluxioURI(src.getName()));
+            asyncCopyLocalPath(pool, src, dst);
+          }
+        } catch (InterruptedException e) {
+          throw new IOException(e);
```

Comment:
should we rethrow as `RuntimeException`? Interruption isn't really an IO exception

Reply:
sg